sanic/__main__.py

```diff
@@ -10,23 +10,70 @@
     parser.add_argument('--port', dest='port', type=int, default=8000)
     parser.add_argument('--workers', dest='workers', type=int, default=1, )
     parser.add_argument('--debug', dest='debug', action="store_true")
+    parser.add_argument('--no_restart', dest='no_restart', action="store_true")
     parser.add_argument('module')
     args = parser.parse_args()
 
     try:
-        module_parts = args.module.split(".")
-        module_name = ".".join(module_parts[:-1])
-        app_name = module_parts[-1]
-
-        module = import_module(module_name)
-        app = getattr(module, app_name, None)
-        if type(app) is not Sanic:
-            raise ValueError("Module is not a Sanic app, it is a {}.  "
-                             "Perhaps you meant {}.app?"
-                             .format(type(app).__name__, args.module))
-
-        app.run(host=args.host, port=args.port,
-                workers=args.workers, debug=args.debug)
+        if args.debug and not args.no_restart:
+            import os
```

Comment:
in the class LiveLoadingHandler, can we add a method on_modified, on which we can listen the modified event and get the src_path, and the just reload the module of src_path

just like that 

    def on_modified(self, event):
        module = path and _find(event.src_path)
        if hasattr(module, '__spec__'):
            from importlib import reload
            app.router.reload()
            reload(module)
            reload(sanic)

def _find(path):
    try:
        return import_module(path)
    except (ImportError, TypeError):
        pass
    path = '.'.join(path.split('.')[:-1])
    if '.' in path:
        return _find(path)
    return path